godtools/App/Features/ToolDetails/ToolDetailViewModel.swift

```diff
@@ -208,20 +211,29 @@ class ToolDetailViewModel: NSObject, ToolDetailViewModelType {
     
     private func handleTranslationManifestDownloaded(result: Result<TranslationManifestData, TranslationDownloaderError>) {
         
+        let tips: [String: Tip]
+        let emptyTips: [String: Tip] = Dictionary()
         let hidesLearnToShareButton: Bool
         
         switch result {
         
         case .success(let translationManifest):
             
-            let manifest: MobileContentXmlManifest = MobileContentXmlManifest(translationManifestData: translationManifest)
-            
-            hidesLearnToShareButton = manifest.tips.isEmpty
+            let result: Result<Manifest, Error> = mobileContentParser.parse(translationManifestData: translationManifest)
             
+            switch result {
+            case .success(let manifest):
+                tips = manifest.tips
```

Comment:
not sure how you are using `tips` elsewhere, but there is a `manifest.findTip(name)` for finding a specific tip model. I'm not sure if this would help with anything in your logic

Reply:
nm, this is tool details where you just need to know if tips exist or not.

Reply:
can't you just do:
```
hidesLearnToShareButton = manifest.tips.isEmpty
```

Reply:
so, I was thinking you didn't need to track intermediary objects at all, your logic could be something as simple as:

```swift
let hidesLearnToShareButton: Bool

let result = parseManifest()
switch result {
    case .success(let manifest):
        hidesLearnToShareButton = manifest.tips.isEmpty
    case .failure(let error):
        hidesLearnToShareButton = true
}
```

Reply:
Hey @frett I like that approach.  I'll include that.